x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/middleware.ts

```diff
@@ -75,57 +75,63 @@ const createTrustedAppsListResourceStateChangedAction = (
   payload: { newState },
 });
 
-const refreshListIfNeeded = async (
+const refreshList = async (
   store: ImmutableMiddlewareAPI<TrustedAppsListPageState, AppAction>,
-  trustedAppsService: TrustedAppsService
+  trustedAppsService: TrustedAppsService,
+  query?: string
 ) => {
-  if (needsRefreshOfListData(store.getState())) {
+  store.dispatch(
+    createTrustedAppsListResourceStateChangedAction({
+      type: 'LoadingResourceState',
+      // need to think on how to avoid the casting
+      previousState: getListResourceState(store.getState()) as Immutable<
+        StaleResourceState<TrustedAppsListData>
+      >,
+    })
+  );
+
+  try {
+    const pageIndex = getCurrentLocationPageIndex(store.getState());
+    const pageSize = getCurrentLocationPageSize(store.getState());
+    const filter = query !== undefined ? query : getCurrentLocationFilter(store.getState());
+
+    const response = await trustedAppsService.getTrustedAppsList({
+      page: pageIndex + 1,
+      per_page: pageSize,
+      kuery: parseQueryFilterToKQL(filter, SEARCHABLE_FIELDS) || undefined,
+    });
+
     store.dispatch(
       createTrustedAppsListResourceStateChangedAction({
-        type: 'LoadingResourceState',
-        // need to think on how to avoid the casting
-        previousState: getListResourceState(store.getState()) as Immutable<
-          StaleResourceState<TrustedAppsListData>
-        >,
+        type: 'LoadedResourceState',
+        data: {
+          items: response.data,
+          pageIndex,
```

Comment:
I know this was there before, but it should be `pageIndex + 1` no?

Reply:
Not at all, trusted apps is working with index - 1. I think @paul-tavares has talked about this before.

Reply:
Yeah, we have inconsistency in our api as far as paging goes. This particular one is still zero-based, which is also the way its represented in the URL, so no need to math here :)